### DISCUSSION TITLE
Error: Unhandled error: SyntaxError: Invalid or unexpected token

### DISCUSSION THREAD

**Discussion by JawadZein** [community/community#64167] [Programming Help]
### Select Topic Area

Question

Hi GitHub Support,

In this block of code I'm getting an error in GitHub workflow

### Body

name: Create GitHub Issue
uses: actions/github-script@v6
with:
github-token: ${{ secrets.GitHub_TOKEN }}
script: |
const fs = require('fs');
const emailContents = fs.readFileSync('/home/runner/work/demo_test/demo_test/scripts/email_contents.txt', 'utf8');
const issueTitle = 'New Issue from Email';
const issueBody = 'emailContents';

      async function createGitHubIssue() {
       # Create an issue using the GitHub API
      const response = await github.issues.create({
      owner: 'context.repo.owner',
      repo: 'context.repo.repo',
      title: 'issueTitle',
      body: 'issueBody'
      });           
      console.log('Issue created:', response.data);
      }
      createGitHubIssue()
      user-agent: actions/github-script
      result-encoding: json
      debug: false           
      retries: 0
      retry-exempt-status-codes: 400,401,403,404,422
Error :

Run actions/github-script@v6
SyntaxError: Invalid or unexpected token
at new AsyncFunction ()
at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15143:16)
at main (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:152[36](https://github.com/JawadZein/demo_test/actions/runs/5894146134/job/15987150221#step:6:37):26)
at /home/runner/work/_actions/actions/github-script/v6/dist/index.js:15217:1
at /home/runner/work/_actions/actions/github-script/v6/dist/index.js:15268:3
at Object. (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15271:12)
at Module._compile (node:internal/modules/cjs/loader:1198:14)
at Object.Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
at Module.load (node:internal/modules/cjs/loader:1076:32)
Error: Unhandled error: SyntaxError: Invalid or unexpected token
at Function.Module._load (node:internal/modules/cjs/loader:911:12)

Thanks

**Comment by nirav-gajera** [comment]
The error message "SyntaxError: Invalid or unexpected token" means that there is a syntax error in the JavaScript code. The most likely cause of the error is that the `issueBody` variable is not defined before it is used.

To fix the error, you need to define the `issueBody` variable before you use it. You can do this by adding the following line of code to the script:

```
let issueBody = 'emailContents';
```

This will define the `issueBody` variable and assign it the value of the `emailContents` variable.

Once you have defined the `issueBody` variable, the error should be resolved.

ex:
```

const fs = require('fs');
const emailContents = fs.readFileSync('/home/runner/work/demo_test/demo_test/scripts/email_contents.txt', 'utf8');

let issueBody = 'emailContents';

async function createGitHubIssue() {
  # Create an issue using the GitHub API
  const response = await github.issues.create({
  owner: 'context.repo.owner',
  repo: 'context.repo.repo',
  title: 'issueTitle',
  body: issueBody
  });           
  console.log('Issue created:', response.data);
  }
  createGitHubIssue()


```

**Comment by JawadZein** [comment]
Thank you for your response!
Your previous suggestion it works but now I'm facing another error in this block 


async function createGitHubIssue() {
              
              const response = await octokit.issues.create({
              owner: context.repo.owner,
              repo: context.repo.repo,
              title: issueTitle,
              body: issueBody
              });           
              console.log('Issue created:', response.data);
            }
            createGitHubIssue()


Error!

Error: Unhandled error: ReferenceError: octokit is not defined

Run actions/github-script@v6
  with:
    github-token: ***
    script: const fs = require('fs');
  const emailContents = fs.readFileSync('/home/runner/work/demo_test/demo_test/scripts/email_contents.txt', 'utf8');
  const issueTitle = 'New Issue from Email';
  let issueBody = emailContents;
  
  async function createGitHubIssue() {
    
    const response = await octokit.issues.create({
    owner: context.repo.owner,
    repo: context.repo.repo,
    title: issueTitle,
    body: issueBody
    });           
    console.log('Issue created:', response.data);
  }
  createGitHubIssue()
  
    debug: false
    user-agent: actions/github-script
    result-encoding: json
    retries: 0
    retry-exempt-status-codes: 400,401,403,404,4[2](https://github.com/JawadZein/demo_test/actions/runs/5900543076/job/16004992966#step:6:2)2
  env:
    pythonLocation: /opt/hostedtoolcache/Python/[3](https://github.com/JawadZein/demo_test/actions/runs/5900543076/job/16004992966#step:6:3).11.[4](https://github.com/JawadZein/demo_test/actions/runs/5900543076/job/16004992966#step:6:4)/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.11.4/x64/lib
ReferenceError: octokit is not defined
    at createGitHubIssue (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:1[5](https://github.com/JawadZein/demo_test/actions/runs/5900543076/job/16004992966#step:6:5)143:1[6](https://github.com/JawadZein/demo_test/actions/runs/5900543076/job/16004992966#step:6:6)), <anonymous>:10:20)
    at eval (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15143:16), <anonymous>:18:1)
    at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15144:12)
    at main (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15236:26)
    at /home/runner/work/_actions/actions/github-script/v6/dist/index.js:1521[7](https://github.com/JawadZein/demo_test/actions/runs/5900543076/job/16004992966#step:6:7):1
**Error: Unhandled error: ReferenceError: octokit is not defined**
    at /home/runner/work/_actions/actions/github-script/v6/dist/index.js:1526[8](https://github.com/JawadZein/demo_test/actions/runs/5900543076/job/16004992966#step:6:8):3
    at Object.<anonymous> (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15271:12)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:115[9](https://github.com/JawadZein/demo_test/actions/runs/5900543076/job/16004992966#step:6:9):[10](https://github.com/JawadZein/demo_test/actions/runs/5900543076/job/16004992966#step:6:10))
    at Module.load (node:internal/modules/cjs/loader:981:[32](https://github.com/JawadZein/demo_test/actions/runs/5900543076/job/16004992966#step:6:33))

**Comment by nirav-gajera** [comment]
you need to import the octokit module into your script. 

`const octokit = require('octokit');`

This will import the octokit module and make the octokit variable available in your script.

Once you have imported the octokit module, you can use it to interact with the GitHub API.

ex: 
```

const fs = require('fs');
const emailContents = fs.readFileSync('/home/runner/work/demo_test/demo_test/scripts/email_contents.txt', 'utf8');
const issueTitle = 'New Issue from Email';
let issueBody = emailContents;

const octokit = require('octokit');

async function createGitHubIssue() {

const response = await octokit.issues.create({
owner: context.repo.owner,
repo: context.repo.repo,
title: issueTitle,
body: issueBody
});

console.log('Issue created:', response.data);
}
createGitHubIssue()
```

**Reply by JawadZein** [reply]
Updated the code facing this Error: Cannot find module 'octokit'
Require stack:
- /home/runner/work/_actions/actions/github-script/v6/dist/index.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:9[33](https://github.com/JawadZein/demo_test/actions/runs/5900907375/job/16006056388#step:6:34):15)
    at resolve (node:internal/modules/cjs/helpers:108:19)
    at Object.apply (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15188:43)
    at eval (eval at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15143:16), <anonymous>:7:17)
    at callAsyncFunction (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15144:12)
    at main (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:152[36](https://github.com/JawadZein/demo_test/actions/runs/5900907375/job/16006056388#step:6:37):26)
    at /home/runner/work/_actions/actions/github-script/v6/dist/index.js:15217:1
    at /home/runner/work/_actions/actions/github-script/v6/dist/index.js:15268:3
    at Object.<anonymous> (/home/runner/work/_actions/actions/github-script/v6/dist/index.js:15271:12)
    at Module._compile (node:internal/modules/cjs/loader:1105:14) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/runner/work/_actions/actions/github-script/v6/dist/index.js'
  ]
}
Error: Unhandled error: Error: Cannot find module 'octokit'
Require stack:
- /home/runner/work/_actions/actions/github-script/v6/dist/index.js

**Reply by nirav-gajera** [reply]
you need to install the octokit module on the runner.You can do this by adding the following line to the dependencies section of your workflow file: 
```
 dependencies:
  octokit: ^1.5.1
```

Once you have added the dependency, you need to run the setup action to install the module on the runner. The setup action is a built-in action that is used to install dependencies on the runner.

ex:

```
name: Create GitHub Issue

on:
push:
branches:
- main

jobs:
create-issue:
runs-on: ubuntu-latest

steps:
  - uses: actions/checkout@v2

  - uses: actions/setup-node@v1
    with:
      node-version: '16'

  - name: Install dependencies
    run: npm install

  - name: Create GitHub Issue
    run: |
      const fs = require('fs');
      const emailContents = fs.readFileSync('/home/runner/work/demo_test/demo_test/scripts/email_contents.txt', 'utf8');
      const issueTitle = 'New Issue from Email';
      let issueBody = emailContents;

      const octokit = require('octokit');

      async function createGitHubIssue() {

        const response = await octokit.issues.create({
          owner: context.repo.owner,
          repo: context.repo.repo,
          title: issueTitle,
          body: issueBody
        });           
        console.log('Issue created:', response.data);
      }
      createGitHubIssue()
```

**Comment by JawadZein** [comment]
Hi when using this code 

name: Run email-to-github script

on:
  push:
    branches:
      - main  # Change this to the appropriate branch name
           
jobs:
  process-email:
    runs-on: ubuntu-latest

    steps:
      - name: Checkout code
        uses: actions/checkout@v2

      - name: check
        uses: actions/setup-node@v1
        with:
          node-version: '16'

      - name: Set up Python
        uses: actions/setup-python@v2
        with:
          python-version: 3.x

      - name: Install dependencies
        run: |
          npm install @octokit/action
          python -m pip install --upgrade pip
          pip install pyyaml
          npm install @octokit/core@5.0.0
          npm install @actions/github
          npm install @octokit/auth-action
          
      - name: Run email-to-github script
        run: python "/home/runner/work/demo_test/demo_test/scripts/email-to-github-script.py"

      - name: Create GitHub Issue
        uses: actions/github-script@v6
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          script: |
            const fs = require('fs');
            const emailContents = fs.readFileSync('/home/runner/work/demo_test/demo_test/scripts/email_contents.txt', 'utf8');
            const issueTitle = 'New Issue from Email';
            let issueBody = emailContents;
           
            const { Octokit } = require("@octokit/action");
            const octokit = new Octokit();
            async function createGitHubIssue() {
              
              const response = await octokit.issues.create({
              owner: context.repo.owner,
              repo:  context.repo.repo,
              title: issueTitle,
              body: issueBody
              });           
              console.log('Issue created:', response.data);
            }
            createGitHubIssue()


Getting this new error 

Error: Unhandled error: Error: [@octokit/auth-action] `GITHUB_TOKEN` variable is not set. It must be set on either `env:` or `with:`. See https://github.com/octokit/auth-action.js#createactionauth

**Comment by JawadZein** [comment]
This was the block of code that worked 


async function createGitHubIssue() {
              for (const emailInfo of emailData) {
              const issueTitle = 'New Issue from Email: ' + emailInfo.subject;
              const issueBody = `**Subject:** ${emailInfo.subject}\n**From:** ${emailInfo.from}\n**Body:**\n${emailInfo.body}`;
              const issueLabels = ['ToDo'];
              const response = await octokit.issues.create({
              owner: context.repo.owner,
              repo:  context.repo.repo,
              title: issueTitle,
              body: issueBody,
              labels: issueLabels
              });           
              console.log('Issue created:', response.data);
            }
            }
            createGitHubIssue()
